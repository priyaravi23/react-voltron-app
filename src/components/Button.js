import React from 'react';
import {func, string, bool} from 'prop-types';
import {EMPTY_FUNC} from "../utils/utils";

import './Button.scss';

const Button = props => {
    const {onClick = EMPTY_FUNC, text, disabled, className} =  props;
    const handleClick = disabled ? EMPTY_FUNC : e => {
        e.preventDefault();
        onClick(e);
    };
    const classNames = 'btn ' + ['primary', 'disabled']
        .filter(c => props[c])
        .concat(className)
        .join(' ');
    return (
        <a className={classNames} href={'#'} onClick={handleClick}>
            {text}
        </a>
    );
};

Button.propTypes = {
    onClick: func,
    text: string,
    disabled: bool,
    primary: bool,
    className: string
};

export default Button;