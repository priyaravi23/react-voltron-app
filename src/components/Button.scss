@import "../variables";

a.btn {
  border: 1px solid $primary-color;
  border-radius: 0.3em;
  padding: 0.35em 0.5em;
  text-decoration: none;
  color: $primary-color;
  transition: color 100ms ease;
  &, &.primary {
    &:active {
      background-color: $secondary-color;
      border-color: $secondary-color;
      color: white;
    }
  }
}

a.btn.primary {
  background-color: $primary-color;
  color: white;
}