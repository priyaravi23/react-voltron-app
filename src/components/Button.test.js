import React from 'react'
import Button from './Button';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() });

describe('<Button /> shallow rendering', () => {
    it('updates classNames based on button click', () => {
        const wrapper = shallow(<Button primary={true}/>);
        expect(wrapper.find('.btn').hasClass('primary')).toBe(true);
    });

    it('matches the snapshot', () => {
        const tree = shallow(<Button />);
        expect(toJson(tree)).toMatchSnapshot();
    });
});