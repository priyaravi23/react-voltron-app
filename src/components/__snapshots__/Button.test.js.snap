// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> shallow rendering matches the snapshot 1`] = `
<a
  className="btn "
  href="#"
  onClick={[Function]}
/>
`;
